### PR TITLE
Fixing error when labelling operators

### PR DESCRIPTION
**Pull request by ramperher**
This fix intends to solve a problem we are having in some daily jobs like this one:
https://www.distributed-ci.io/jobs/dbeabe85-6aab-4e9d-a012-636cf6a2ec36/jobStates#71726004-54dd-4ecf-92c2-2c6b0b3df7c0:file66
It may happen that installplans for a given operator is empty or returning more than one element, depending on how example-cnf installation was. Currently, we are only handling one installplan per operator checked, so with this, we will treat these cases in which we can have 0 or more than 1 installplans per operator.
This was tested manually in CLI and it works.

### CODE REVIEW

**Comment by ramperher** [comment]
All the jobs have failed in previous steps not related to this change.  I will try to relaunch some more tests tomorrow if needed.

**Comment by ramperher** [comment]
Last job worked with OCP 4.10, which is the OCP version that is failing in the daily jobs regarding this issue. So this can be merged.

**Comment by ramperher** [comment]
@betoredhat including you in the loop as you are the CI Duty person this week :)

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (tonyskapunk) [inline testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:45]: You can use bash built-in array length instead:
```suggestion
if [ ${#INSTALL_PLAN_NAMES[@]} -ne 0 ]; then
```

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (tonyskapunk) [inline testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:48]: You can append to arrays with `+=` e.g.
```suggestion
        SUBSCRIPTION_NAMES+=( $(oc get installplan "${INSTALL_PLAN_NAME}" -n "{{ app_ns }}" -o json | jq -r ".metadata.ownerReferences[].name") )
```
Added spacing to read easier that is an array

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (tonyskapunk) [inline testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:50]: You could include the `unique` function inside your initial jq:
```suggestion
      SUBSCRIPTION_NAMES_JSON=$(echo ${SUBSCRIPTION_NAMES[@]} | jq -R 'split (" ")| unique')
```

**Review by tonyskapunk — COMMENTED** [review]
Adding some suggestions, but the change LGTM  :+1:

**Review by betoredhat — COMMENTED** [review]
The change looks good to me. 
IMHO if we are using oc for some tasks, maybe getting the subscription names, Install plans, etc could be also using oc instead of querying the  example_cnf_csv_list.resources json multiple times. 

Tony's recommendations worth to be implemented.

Thanks

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (ramperher) [reply testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:45]: If you check the [commits](https://github.com/dci-labs/example-cnf-config/pull/39/commits/27d6e309f99de492e8e3b52fff5e615dd0e20483), I included this in the first version, but I don't know why, Ansible was compaining about this, so that's why I had to change it.

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (ramperher) [reply testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:48]: done

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (ramperher) [reply testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:50]: done

**Comment by ramperher** [comment]
I have included Tony's recommendation excepting the first one. As I explained, that was included in the first commit of this PR, but it was failing in Ansible (not failing directly on CLI), so that I have to move to the current code.

> IMHO if we are using oc for some tasks, maybe getting the subscription names, Install plans, etc could be also using oc instead of querying the example_cnf_csv_list.resources json multiple times.

In this case, I just use `oc` here because it was easier for me in order to manipulate strings and build the JSON array that I needed. Moreover, this info is only used in this task, rather than in other tasks. For that reason, I maintained the other tasks that are not using `oc`, because they are simpler to program and they are also providing information that may be useful in the future if needed, so that's why I would say I would keep them as they are right now. But this is my opinion, of course :)

**Comment by ramperher** [comment]
By the way, to avoid booking a cluster to repeat example-cnf execution, I tested Tony's modification locally and with an Ansible playbook using openshift-performance-addon-operator namespace as example and it worked in both cases:

**Locally**

```
[kni@provisioner.cluster5.dfwt5g.lab ~]$ oc get installplan -n openshift-performance-addon-operator
NAME            CSV                                 APPROVAL    APPROVED
install-wlrx9   performance-addon-operator.v4.7.6   Automatic   true
[kni@provisioner.cluster5.dfwt5g.lab ~]$ INSTALL_PLAN_NAMES=($(oc get installplan -n openshift-performance-addon-operator | grep install- | awk '{ print $1 }'))
[kni@provisioner.cluster5.dfwt5g.lab ~]$ INSTALL_PLAN_NUMBER=$(echo -n ${INSTALL_PLAN_NAMES[@]} | wc -w)
[kni@provisioner.cluster5.dfwt5g.lab ~]$ echo $INSTALL_PLAN_NUMBER 
1
[kni@provisioner.cluster5.dfwt5g.lab ~]$ SUBSCRIPTION_NAMES=()
[kni@provisioner.cluster5.dfwt5g.lab ~]$ SUBSCRIPTION_NAMES+=( $(oc get installplan "${INSTALL_PLAN_NAMES}" -n openshift-performance-addon-operator -o json | jq -r ".metadata.ownerReferences[].name") )
[kni@provisioner.cluster5.dfwt5g.lab ~]$ SUBSCRIPTION_NAMES+=( $(oc get installplan "${INSTALL_PLAN_NAMES}" -n openshift-performance-addon-operator -o json | jq -r ".metadata.ownerReferences[].name") )
[kni@provisioner.cluster5.dfwt5g.lab ~]$ echo ${SUBSCRIPTION_NAMES[@]}
performance-addon-operator-subscription performance-addon-operator-subscription
[kni@provisioner.cluster5.dfwt5g.lab ~]$ SUBSCRIPTION_NAMES_JSON=$(echo ${SUBSCRIPTION_NAMES[@]} | jq -R 'split (" ")| unique')
[kni@provisioner.cluster5.dfwt5g.lab ~]$ echo $SUBSCRIPTION_NAMES_JSON 
[ "performance-addon-operator-subscription" ]
```

**Ansible playbook**

```
  - name: Tag CSVs with autodiscovery labels/annotations to execute test suites
    shell: |
      set -ex

      export KUBECONFIG="/home/dciteam/clusterconfigs-cluster5/kubeconfig"
      INSTALL_PLAN_NAMES=($(oc get installplan -n openshift-performance-addon-operator | grep install- | awk '{ print $1 }'))
      INSTALL_PLAN_NUMBER=$(echo -n ${INSTALL_PLAN_NAMES[@]} | wc -w)
      echo $INSTALL_PLAN_NUMBER
      SUBSCRIPTION_NAMES=()
      SUBSCRIPTION_NAMES+=( $(oc get installplan "${INSTALL_PLAN_NAMES}" -n openshift-performance-addon-operator -o json | jq -r ".metadata.ownerReferences[].name") )
      SUBSCRIPTION_NAMES+=( $(oc get installplan "${INSTALL_PLAN_NAMES}" -n openshift-performance-addon-operator -o json | jq -r ".metadata.ownerReferences[].name") )
      echo ${SUBSCRIPTION_NAMES[@]}
      SUBSCRIPTION_NAMES_JSON=$(echo ${SUBSCRIPTION_NAMES[@]} | jq -R 'split (" ")| unique')
      echo $SUBSCRIPTION_NAMES_JSON
    register: output

  - name: debug output
    debug:
      msg: "{{ output.stdout }}"

...

TASK [Tag CSVs with autodiscovery labels/annotations to execute test suites] **************************************************************************************************************************************
changed: [localhost]


TASK [debug output] ***********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "1\nperformance-addon-operator-subscription performance-addon-operator-subscription\n[ \"performance-addon-operator-subscription\" ]"
}
```

So this should be ready to be merged if you approve it. I will wait for your approval before doing it.

**Review by betoredhat — APPROVED** [review]
LGTM, thanks Ramon

File: testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml
Comment (tonyskapunk) [reply testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:45]: :thinking: This is odd, indeed.  I did a quick test, and seems to be working, I'm using a slightly different entry, but should be equivalent:

```
$ IP=($(oc get installplan -A | awk '/openshift/ {print $1}'))
echo ${#IP[@]}
3
```

Do you have a job where it failed,  we could figure out what's going on.

Comment (tonyskapunk) [reply testpmd/hooks/roles/apply-cnf-cert-config/tasks/install.yml:45]: I figured this out, is due to jinja2 template the "way" to fix the issue with `{#` (jinja commenting) is to build char by char then let the template append it when rendered, but is too much hassle:

```yaml
- name: shell jinja
  hosts: localhost
  gather_facts: false
  tasks:
  - name: shell
    shell: |
      x=( uno dos tres )
      echo ${x}
      echo ${{ "{" }}{{ "#" }}x[@]}
```

**Pull request merged**
